### PR TITLE
feat: migrate lambda module to its own repo

### PR DESCRIPTION
**Pull request by bunchmj**

### CODE REVIEW

**Comment by zack-is-cool** [comment]
> I guess this is reliant on this [defenseunicorns/terraform-aws-uds-lambda#2](https://github.com/defenseunicorns/terraform-aws-uds-lambda/pull/2)

Yeah, once we cut the release on the lambda module one we just update the ref

**Review by zack-is-cool — APPROVED** [review]

File: examples/complete/main.tf [hunk 1]
```diff
@@ -439,7 +439,7 @@ module "password_lambda" {
 
   count = var.enable_bastion ? 1 : 0
 
-  source                          = "../../modules/lambda"
+  source                          = "git::https://github.com/defenseunicorns/terraform-aws-uds-lambda.git?ref=v0.0.2"
```
Comment (zack-is-cool) [inline examples/complete/main.tf:442]:
```suggestion
  source                          = "git::https://github.com/defenseunicorns/terraform-aws-uds-lambda.git//modules/password-rotation?ref=v0.0.2"
```

Comment (zack-is-cool) [reply examples/complete/main.tf:442]: I think it blew up because you need to ref the submodule

**Review by zack-is-cool — APPROVED** [review]

**Review by zack-is-cool — APPROVED** [review]

**Comment by bunchmj** [comment]
Looks like `output "lambda_password_function_arn" ` is commented out in the lambda module repo here:
https://github.com/defenseunicorns/terraform-aws-uds-lambda/blob/cc54c68292b0af882c2a132298f5df2d8a638465/modules/password-rotation/output.tf

...maybe that could be important
```
Error: Error in function call
│ 
│   on outputs.tf line 41, in output "lambda_password_function_arn":
│   41:   value       = try(module.password_lambda[0].lambda_function_arn)
│     ├────────────────
│     │ while calling try(expressions...)
│     │ module.password_lambda[0] is a object
│ 
│ Call to function "try" failed: no expression succeeded:
│ - Unsupported attribute (at outputs.tf:41,46-66)
│   This object does not have an attribute named "lambda_function_arn".
```

**Pull request merged**